### ISSUE TITLE
AWS App Mesh GA additions

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#8102]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Need to diff the AWS App Mesh GA changes and support additions.
Resource tagging is a separate issue: https://github.com/terraform-providers/terraform-provider-aws/issues/8101.

**Comment by ewbankkit** [comment]
Requires [AWS SDK v1.19.3](https://github.com/aws/aws-sdk-go/releases/tag/v1.19.3): https://github.com/terraform-providers/terraform-provider-aws/pull/8098.
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/6707

**Comment by ewbankkit** [comment]
#### [`aws_appmesh_mesh`](https://www.terraform.io/docs/providers/aws/r/appmesh_mesh.html)
Add support for [egress filter rules](https://github.com/aws/aws-app-mesh-roadmap/issues/2):

```hcl
resource "aws_appmesh_mesh" "simple" {
  name = "simpleapp"

  spec {
    egress_filter {
      type = "ALLOW_ALL"
    }
  }
}
```

#### [`aws_appmesh_route`](https://www.terraform.io/docs/providers/aws/r/appmesh_route.html)
Add support for [TCP routing](https://github.com/aws/aws-app-mesh-roadmap/issues/4):

```hcl
resource "aws_appmesh_route" "serviceb" {
  name                = "serviceB-route"
  mesh_name           = "${aws_appmesh_mesh.simple.id}"
  virtual_router_name = "${aws_appmesh_virtual_router.serviceb.name}"

  spec {
    tcp_route {
      action {
        weighted_target {
          virtual_node = "${aws_appmesh_virtual_node.serviceb1.name}"
          weight       = 100
        }
      }
    }
  }
}
```

#### [`aws_appmesh_virtual_node`](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_node.html)

Add support for [access logging](https://github.com/aws/aws-app-mesh-roadmap/issues/18):

```hcl
resource "aws_appmesh_virtual_node" "serviceb1" {
  name                = "serviceBv1"
  mesh_name           = "${aws_appmesh_mesh.simple.id}"

  spec {
    logging {
      access_log {
        file {
          path = "/dev/stdout"
        }
      }
    }
  }
}
```

**Comment by ewbankkit** [comment]
@bcelenza Have I captured all the GA changes here (tagging is covered in a separate issue, https://github.com/terraform-providers/terraform-provider-aws/issues/8101)?

**Comment by bcelenza** [comment]
@ewbankkit Yes, you've captured them well. Looks good to me. :)

**Comment by bflad** [comment]
[These enhancements](https://github.com/terraform-providers/terraform-provider-aws/issues/8102#issuecomment-477689334) has been merged and will release with version 2.6.0 of the Terraform AWS Provider later today. 👍

**Comment by bflad** [comment]
This has been released in [version 2.6.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#260-april-10-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
